src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1456,7 +1761,12 @@ private new void VisitCollectionElementInitializer(BoundCollectionElementInitial
         {
             // Note: we analyze even omitted calls
             VisitArguments(node, node.Arguments, refKindsOpt: default, node.AddMethod, node.ArgsToParamsOpt, node.Expanded);
-            SetUnknownResultNullability();
+            if (node.ImplicitReceiverOpt != null)
+            {
+                Debug.Assert(node.ImplicitReceiverOpt.Kind == BoundKind.ImplicitReceiver);
+                SetAnalyzedNullability(node.ImplicitReceiverOpt, new VisitResult(node.ImplicitReceiverOpt.Type, NullableAnnotation.NotAnnotated, NullableFlowState.NotNull));
+            }
+            SetUnknownResultNullability(node);
```

Comment:
>SetUnknownResultNullability [](start = 12, length = 27)

Is this going to record an analyzed nullability again (clashing with `SetAnalyzedNullability` call above)?

Reply:
No. That's on the implicit receiver, this on the collection element initializer.

---
In reply to: [265704071](https://github.com/dotnet/roslyn/pull/34017#discussion_r265704071) [](ancestors = 265704071)